src/Kirel.Identity.Core/Services/KirelAuthenticationService.cs

```diff
@@ -1,158 +1,47 @@
-        var lifeTime = Convert.ToDouble(lifetime);
-        var jwt = new JwtSecurityToken(
-            AuthOptions.Issuer,
-            AuthOptions.Audience,
-            notBefore: now,
-            claims: claims.Claims,
-            expires: now.AddMinutes(lifeTime),
-            signingCredentials: new SigningCredentials(AuthOptions.GetSymmetricSecurityKey(AuthOptions.Key), SecurityAlgorithms.HmacSha256));
-        return new JwtSecurityTokenHandler().WriteToken(jwt);
-    }
     
     /// <summary>
-    /// Method for add new user to identity db
+    /// Constructor for KirelAuthenticationService
     /// </summary>
-    /// <param name="createDto">RegistrationRequestDto</param>
-    /// <returns>New application user</returns>
-    public virtual async Task Register(TRegisterDto createDto)
-    {
-        var appUser = Mapper.Map<TUser>(createDto);
-        var result = await UserManager.CreateAsync(appUser);
-        if(!result.Succeeded)  throw new AuthenticationException("Failed to create new user");
-        var passwordResult = await UserManager.AddPasswordAsync(appUser, createDto.Password);
-        if (!passwordResult.Succeeded)
-        {
-            await UserManager.DeleteAsync(appUser);
-            throw new AuthenticationException("Failed to add password");
-        }
-    }
-
-    private async Task<JwtTokenDto> GenerateTokensPair(TUser user)
+    /// <param name="userManager"></param>
+    public KirelAuthenticationService(UserManager<TUser> userManager)
     {
-        var userClaims = await GetUserIdentityClaims(user);
-        if (userClaims == null) throw new AuthenticationException($"User with login {user.UserName} is not found");
-        
-        var accessToken = CreateJwtToken(userClaims, AuthOptions.AccessLifetime);
-
-        var claimsList = new List<Claim>
-        {
-            new(ClaimTypes.Name, user.UserName),
-            new(ClaimTypes.Role, "RefreshToken")
-        };
-        var refreshClaims = new ClaimsIdentity(claimsList, "Token", ClaimsIdentity.DefaultNameClaimType, ClaimsIdentity.DefaultRoleClaimType);
-        var refreshToken = CreateJwtToken(refreshClaims, AuthOptions.RefreshLifetime);
-
-        return new JwtTokenDto()
-        {
-            AccessToken = accessToken,
-            RefreshToken = refreshToken
-        };
+        UserManager = userManager;
     }
     
     /// <summary>
-    /// Method for getting JWT token
+    /// Provides the ability to get the identity user after checking the login and password
     /// </summary>
     /// <param name="login">User login</param>
     /// <param name="password">User password</param>
-    public virtual async Task<JwtTokenDto> GetJwtToken(string login, string password)
+    /// <returns>User class instance</returns>
+    /// <exception cref="AuthenticationException">Returned if the user is not found or if the password is incorrect</exception>
+    public async Task<TUser> LoginAndPassword(string login, string password)
```

Comment:
Since we use verbs in functions names, i think it will be better to name this function LoginByPassword